mitodl/ocw-hugo-themes
# improve start course, start www, start fields

#### Pre-Flight checklist


#### What are the relevant tickets?
#974 

#### What's this PR do?
This PR replace `yarn start:course`, etc, with `yarn start course`, etc.

Additionally, the new `yarn start` commands will clone dependencies as needed and can more easily be passed cli arguments.

```
➜ yarn start --help
Usage: start [options] [command]

Start ocw-hugo-themes development servers.

Options:
  -h, --help                  display help for command

Commands:
  course [options] [shortid]  Start webpack + hugo dev servers for an ocw course.
  www [options]               Start webpack + hugo dev servers for ocw-www (homepage).
  fields [options]            Start webpack + hugo dev servers for mit-fields.
  help [command]              display help for command
```

#### How should this be manually tested?
1. Delete or rename your ocw-hugo-projects folder.
2. Delete or rename your site content folders.
3. Run `yarn start course`. *you should be prompted about downloading `ocw-hugo-projects` and site content repo.*
4. Run `yarn start www`. *you should be prompted about downloading site content repo.*
5. Run `yarn start fields`. *you should be prompted about downloading site content repo.*
6. Run `yarn start` by itself, and also `yarn start course --help`, `yarn start www --help`, `yarn start fields --help`
7. Try new CLI options:
    - `yarn start course 8.01sc-fall-2016` (or any other course short id).
    -  run a difference config file, e.g., `yarn start course --config ~/dev/ocw-hugo-projects/ocw-course/config.yaml`
    - The CLI options take their default values from environment variables, so if you're constantly using the v1 theme (for some reason) you could set `COURSE_HUGO_CONFIG_PATH` to v1 and the cli should respect it.

Test any other CLI variants you want to try.
